### PR TITLE
Add UT for remedy util function and eventhandler

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

- Add UT for remedy util function.
- Add UT for remedy controller event handler

**Which issue(s) this PR fixes**:
Part of #4658

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
cc @RainbowMango

File: pkg/controllers/remediation/eventhandlers_test.go [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+			case "Generic":
+				genericEvent := event.GenericEvent{Object: tt.args.obj}
+				h.Generic(context.TODO(), genericEvent, queue)
+			default:
+				t.Errorf("no support operation %v", tt.args.operation)
+				return
+			}
+
+			if got := queue.Len(); got != tt.wantQLen {
+				t.Errorf("clusterEventHandler Create() process queue length = %v, want %v", got, tt.wantQLen)
```
Comment (RainbowMango) [inline pkg/controllers/remediation/eventhandlers_test.go:164]: This log seems incorrect, as here not just focus on `Create`.

File: pkg/controllers/remediation/eventhandlers_test.go [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+				t.Errorf("no support operation %v", tt.args.operation)
+				return
+			}
+
+			got := 0
+			for range clusterChan {
+				got++
+			}
+			if got != tt.wantChanLen {
+				t.Errorf("remedyEventHandler Create() process chan length = %v, want %v", got, tt.wantChanLen)
```
Comment (RainbowMango) [inline pkg/controllers/remediation/eventhandlers_test.go:368]: ditto.

File: pkg/controllers/remediation/eventhandlers_test.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply pkg/controllers/remediation/eventhandlers_test.go]: updated, thx~

File: pkg/controllers/remediation/eventhandlers_test.go [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+			case "Generic":
+				genericEvent := event.GenericEvent{Object: tt.args.obj}
+				h.Generic(context.TODO(), genericEvent, queue)
+			default:
+				t.Errorf("no support operation %v", tt.args.operation)
+				return
+			}
+
+			if got := queue.Len(); got != tt.wantQLen {
+				t.Errorf("clusterEventHandler %v() process queue length = %v, want %v", tt.args.operation, got, tt.wantQLen)
```
Comment (RainbowMango) [inline pkg/controllers/remediation/eventhandlers_test.go:164]: One more thing, actually we don't need to log the target function name here, as the go test would print it out, like:
```
=== RUN   Test_remedyEventHandler
=== RUN   Test_remedyEventHandler/create_event:_remedy_with_clusterAffinity
=== RUN   Test_remedyEventHandler/create_event:_remedy_with_nil_clusterAffinity
=== RUN   Test_remedyEventHandler/update_event:_old_and_new_remedy_all_have_nil_clusterAffinity
=== RUN   Test_remedyEventHandler/update_event:_one_of_the_old_and_new_remedy_have_nil_clusterAffinity
=== RUN   Test_remedyEventHandler/update_event:_the_old_and_new_remedy_clusterAffinity_changed
=== RUN   Test_remedyEventHandler/delete_event:_the_remedy_with_clusterAffinity
=== RUN   Test_remedyEventHandler/delete_event:_the_remedy_with_nil_clusterAffinity
=== RUN   Test_remedyEventHandler/generic_event
--- PASS: Test_remedyEventHandler (0.00s)
    --- PASS: Test_remedyEventHandler/create_event:_remedy_with_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/create_event:_remedy_with_nil_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/update_event:_old_and_new_remedy_all_have_nil_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/update_event:_one_of_the_old_and_new_remedy_have_nil_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/update_event:_the_old_and_new_remedy_clusterAffinity_changed (0.00s)
    --- PASS: Test_remedyEventHandler/delete_event:_the_remedy_with_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/delete_event:_the_remedy_with_nil_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/generic_event (0.00s)
```
It's no big deal, we didn't pay attention to this before.  clean it or not, it's up to you.

Comment (XiShanYongYe-Chang) [reply pkg/controllers/remediation/eventhandlers_test.go]: That makes sense. I'll change it.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**